src/js/timeManage.js

```diff
@@ -0,0 +1,152 @@
+import 'bootstrap/dist/css/bootstrap.css';
+
+/////init/////
+var t = TrelloPowerUp.iframe();
+
+var members = t.arg("members").members;
+members.forEach(member => {
+    var optionText = member.fullName;
+    var optionValue = member.fullName;
+    $('#members').append(`<option value="${optionValue}"> 
+                                       ${optionText} 
+                                  </option>`);
+});
+
+//TODO default date to Today ==> use of moment.js?
+// $('#dateSpent').attr("value", DateTime.Today.ToString("yyyy-MM-dd"));
+
+/////utils card/////
+
+function addTimeToTotalSpent(value, date, member) {
+    //TODO check if all value are set
+
+    return new Promise((resolve) => {
+        t.get('card', 'shared', 'timeTrack').then(function (data) {
+            if (typeof data == 'undefined') {
+                data = {
+                    logs: new Array
+                }
+            }
+            data.logs.push({
+                date: date,
+                timeSpent: value,
+                member: member
+            });
+            t.set('card', 'shared', 'timeTrack', data).then(function () {
+                resolve();
+            });
+        }, function (error) {
+            console.log('error get timeTrack in addTimeToTotalSpent');
+        });
+    });
+}
+
+function calculTotalTimeSpent() {
+    return new Promise((resolve) => {
+        t.get('card', 'shared', 'timeTrack').then(function (data) {
+            var totalTimeSpent = 0;
+            if (typeof data !== 'undefined') {
+                data.logs.forEach(log => {
+                    totalTimeSpent += parseInt(log.timeSpent);
+                });
+            } else {
+                data = {
+                    logs: new Array
+                }
+            }
+            resolve(totalTimeSpent);
+        }, function (error) {
+            console.log('error get timeTrack in calculTotalTimeSpent');
+        });
+    });
+}
```

Comment:
```suggestion
async function calculTotalTimeSpent() {
  let data = await t.get('card', 'shared', 'timeTrack');
  return data ? data.logs.reduce((acc, log) => acc + parseInt(log.timeSpent, 10)) : 0;
}
```

Reply:
Also may want to rename to `calculateTotalTimeSpent`

Reply:
Name Changed and function updated.